Purchases/Public/RCPurchases.m

```diff
@@ -1039,7 +1039,26 @@ - (void)storeKitWrapper:(RCStoreKitWrapper *)storeKitWrapper
             }
             break;
         }
-        case SKPaymentTransactionStateDeferred:
+        case SKPaymentTransactionStateDeferred: {
+            RCPurchaseCompletedBlock completion = nil;
+            @synchronized (self) {
+                completion = self.purchaseCompleteCallbacks[transaction.payment.productIdentifier];
+            }
+            NSError *pendingError = [NSError errorWithDomain:RCPurchasesErrorDomain
+                                                        code:RCPaymentPendingError
+                                                    userInfo:@{
+                                                        NSLocalizedDescriptionKey: @"The payment is deferred."
+                                                    }];
+            CALL_IF_SET_ON_MAIN_THREAD(completion,
+                                       transaction,
+                                       nil,
+                                       pendingError,
+                                       transaction.error.code == SKErrorPaymentCancelled);
+            @synchronized (self) {
+                self.purchaseCompleteCallbacks[transaction.payment.productIdentifier] = nil;
```

Comment:
I wonder if we should start checking if `transaction`, `transaction.payment` and `transaction.payment.productIdentifier` aren't nil before doing this. otherwise we could run into issues like #https://github.com/RevenueCat/purchases-ios/issues/279

On another note, we might not need to make two locks - why not unify the synchronize calls? 
```objective-c
@synchronized (self) {
    completion = self.purchaseCompleteCallbacks[transaction.payment.productIdentifier];
    self.purchaseCompleteCallbacks[transaction.payment.productIdentifier] = nil;
}
```

the block should be copied anyway, right?

Reply:
Very good point! Regarding the blocks, yes, I think we can combine the synchronized blocks